src/lib.rs

```diff
@@ -63,6 +63,7 @@
 //!
 #![deny(missing_debug_implementations)]
 #![warn(missing_docs, unused_extern_crates, trivial_casts, unused_results)]
+#![allow(clippy::unit_arg)]
```

Comment:
Do you remember why this was required? It seems strange.

Reply:
This is something I might look into deeper, but [`unit_arg`](https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg) was triggered by all those function calls with `Default::default()` as the last parameter, which resolved to a unit value. An example of such a call is [here](https://github.com/Enet4/nifti-rs/blob/master/src/object.rs#L61).

I currently find this to be a false positive, because the case for why this lint exists does not really apply. The point of `Default::default()` is to generalize over whichever associated type `Options` is, and the recommendation made by the lint does not even make any sense.

```none
warning: passing a unit value to a function
  --> src/object.rs:67:13
   |
67 |             InMemNiftiObject::from_file_impl(path, GzDecoder::new(file), Default::default())
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::unit_arg)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg
help: move the expression in front of the call and replace it with the unit literal `()`
   |
67 |             Default::default();
68 |             InMemNiftiObject::from_file_impl(path, GzDecoder::new(file), ())
   |
```